torch/csrc/cuda/comm.cpp

```diff
@@ -47,7 +47,13 @@ std::vector<Tensor> broadcast(const Tensor& tensor, IntList devices) {
   {
 #endif
     auto & gpu_type = type.toBackend(type.is_sparse() ? at::kSparseCUDA : at::kCUDA);
-    for (auto device : devices) {
+    if (type.is_cuda()) {
+      tensors.push_back(tensor);
```

Comment:
You didn't check if it's on the right GPU

Reply:
If it's not on the right GPU, an error is thrown at line 32.